osquery/events/darwin/event_taps.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <ApplicationServices/ApplicationServices.h>
+
+#include <osquery/flags.h>
+
+#include "osquery/events/darwin/event_taps.h"
+
+namespace osquery {
+
+/// Flag that turns the eventing system for event taps on or off
+FLAG(bool,
+     disable_event_tapping,
+     true,
+     "Disable receiving and subscribing to events from the event taps "
+     "subsystem");
+
+REGISTER(EventTappingEventPublisher, "event_publisher", "event_tapping");
+
+CGEventRef EventTappingEventPublisher::eventCallback(CGEventTapProxy proxy,
+                                                     CGEventType type,
+                                                     CGEventRef event,
+                                                     void* refcon) {
+  auto ec = createEventContext();
+  EventFactory::fire<EventTappingEventPublisher>(ec);
+  return event;
+}
+
+Status EventTappingEventPublisher::setUp() {
+  if (FLAGS_disable_event_tapping) {
+    return Status(1, "Publisher disabled via configuration");
+  }
+  return Status(0);
+}
+
+void EventTappingEventPublisher::configure() {
+  WriteLock lock(mutex_);
+  restart();
+}
+
+void EventTappingEventPublisher::tearDown() {
+  stop();
+
+  WriteLock lock(mutex_);
```

Comment:
I'd recommend nixing this lock and letting the `stop()` method handle everything.